src/v2.rs

```diff
@@ -159,10 +167,16 @@ impl Party {
 
         for key_id in &self.key_ids {
             self.private_keys.insert(*key_id, Scalar::zero());
-
-            for (_sender, s) in &shares[key_id] {
-                self.private_keys
-                    .insert(*key_id, self.private_keys[key_id] + s);
+            if let Some(shares) = private_shares.get(key_id) {
+                for (_sender, s) in shares {
+                    self.private_keys
+                        .insert(*key_id, self.private_keys[key_id] + s);
+                }
```

Comment:
We can simplify this to
```rust
if let Some(shares) = private_shares.get(key_id) {
    let secret = shares.values().fold(Scalar::zero(), |a, b| a + b);
    self.private_keys.insert(*key_id, secret);
```
or if you don't like that then I'd do
```rust
if let Some(shares) = private_shares.get(key_id) {
    let mut secret = Scalar::zero();
    for s in shares.values() {
        secret += s;
    }
    self.private_keys.insert(*key_id, secret);
```

Reply:
Done, in `v1` and `v2`.  I also filed a ticket to `impl Sum`, since that would be even more concise:

https://github.com/Trust-Machines/p256k1/issues/84

Reply:
Thanks to the quick review, we're using `Sum` now